docs/Tutorials/Configure-External-Signer-TLS.md

```diff
@@ -0,0 +1,196 @@
+---
+title: Configure TLS communication with an external signer
+---
+
+# Configure TLS communication
+
+Configure TLS communication with an external signer such as [Web3Signer] which accepts
+connections from clients that use trusted CA certificates or self-signed certificates.
+
+This tutorial configures TLS between Teku and Web3Signer, and use the
+[`keytool`](https://docs.oracle.com/en/java/javase/12/tools/keytool.html) utility to generate
+keystores and the truststore that contain self-signed certificates.
+
+!!! info
+
+    `keytool` is available with the JDK installation, you can also use OpenSSL.
+
+**Prerequisites**:
+
+- [Web3Signer installed](https://docs.web3signer.consensys.net/en/latest/HowTo/Get-Started/Install-Binaries/).
+- [Web3Signer signing key files](https://docs.web3signer.consensys.net/en/latest/HowTo/Use-Signing-Keys/) for
+    validators on the Pyrmont testnet.
+- [Teku Installed](../HowTo/Get-Started/Installation-Options/Install-Binaries.md).
+- [Java `keytool`](https://docs.oracle.com/en/java/javase/12/tools/keytool.html).
+- A running Ethereum 1.0 node such as [Hyperledger Besu], or cloud-based service such as [Infura]
+    synced to the Goerli testnet.
+
+This tutorial connects to the Pyrmont testnet ETH2 testnet, and uses [Infura] to access the
+Goerli ETH1 testnet. You can sign up for a free [Infura] account, or you can start your own
+[ETH1 Goerli node].
+
+## 1. Create keystores
+
+A keystore contains the certificate and private key used to authenticate yourself during TLS
+mutual authentication.
+
+Teku can use either the PKCS12 or JKS keystore type, whereas Web3Signer only uses a PKCS12 keystore.
+
+For each keystore you must create a plain text file containing the password to decrypt the keystore.
+
+### Web3Signer keystore and password file
+
+1. Generate the Web3Signer keystore.
+
+    ```bash
+    keytool -genkeypair -keystore web3signer_keystore.p12 -storetype PKCS12 -storepass changeit -alias web3signer -keyalg RSA -keysize 2048 -validity 109500 -dname "CN=localhost, OU=PegaSys, O=ConsenSys, L=Brisbane, ST=QLD, C=AU" -ext san=dns:localhost,ip:127.0.0.1
+    ```
+
+    !!! info
+
+        `CN` is generally the fully qualified name of Web3Server, you can use `-ext san` to add additional
```

Comment:
`CN` (common name) is generally the fully qualified host name of Web3Server

Reply:
Done